src/parpeoptimization/optimizationProblem.cpp

```diff
@@ -54,6 +54,87 @@ void *getLocalOptimumThreadWrapper(void *optimizationProblemVp) {
     return result;
 }
 
+void optimizationProblemGradientCheckMultiEps(OptimizationProblem *problem,
+                                      int numParameterIndicesToCheck
+                                      ) {
+    // set eps
+    std::vector<double> multi_eps {1e-1, 1e-3, 1e-5, 1e-7, 1e-9};
+
+    // setting the number of parameters to the minimum of
+    // numParamaterIndicesToCheck and dimension of the problem
+    int numParameters = problem->cost_fun_->numParameters();
+    numParameterIndicesToCheck =
+        std::min(numParameterIndicesToCheck, numParameters);
+    // choose random parameters to check
+    std::vector<int> parameterIndices(numParameters); // ask Daniel
+    std::iota(parameterIndices.begin(), parameterIndices.end(), 0);
+    std::random_device rd;
+    std::mt19937 g(rd());
+
+    optimizationProblemGradientCheckMultiEps(problem, parameterIndices, multi_eps);
+}
+
+void optimizationProblemGradientCheckMultiEps(OptimizationProblem *problem,
+                                              gsl::span<const int> parameterIndices,
+                                              gsl::span<double> multi_eps){
+    //get a random theta
+    std::vector<double> theta(problem->cost_fun_->numParameters());
+    problem->fillInitialParameters(theta);
+    double fc = 0; // f(theta)
+    //evaluate the objective function at theta and get analytical gradient
+    std::vector<double> gradient(theta.size());
+    problem->cost_fun_->evaluate(theta, fc, gradient);
+
+    std::vector<double> thetaTmp(theta);
+    // loop through indices, then loop trough epsilons
+    for(int curInd : parameterIndices) {
+        double eps_best=0.0;
```

Comment:
```suggestion
        double eps_best = 0.0;
```
(same for others below, please use consistent whitespace. your IDE should be able to take care of that.)